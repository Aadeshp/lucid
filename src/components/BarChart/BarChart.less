.lucid-BarChart {
	display: inline-block;

	.axis {
		font-size: @font;

		line,
		path {
			fill: none;
			stroke: @color-black;
		}
	}

	&-x-axis-label {
		text-anchor: middle;
	}

	&-y-axis-label {
		text-anchor: middle;
		transform: rotate(-90deg);
	}

	&-bar {
		fill: @color-primary;
	}
}
