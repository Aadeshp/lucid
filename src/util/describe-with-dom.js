function describeWithDOM(name, func) {
	describe(`${name} (functional tests using ${window.realOrFake})`, func);
}

describeWithDOM.only = function (name, func) {
	describe.only(`${name} (functional tests using ${window.realOrFake})`, func);
};

export default describeWithDOM;
