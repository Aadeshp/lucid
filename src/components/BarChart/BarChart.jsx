import _ from 'lodash';
import React from 'react';
import { lucidClassNames } from '../../util/style-helpers';
import { createClass } from '../../util/component-types';
import {
	maxByFields,
	maxByFieldsStacked,
} from '../../util/chart-helpers';
import d3Scale from 'd3-scale';

import Axis from '../Axis/Axis';
import AxisLabel from '../AxisLabel/AxisLabel';
import Bars from '../Bars/Bars';

const cx = lucidClassNames.bind('&-BarChart');

const {
	any,
	arrayOf,
	func,
	number,
	object,
	shape,
	string,
	array,
	bool,
} = React.PropTypes;

/**
 * {"categories": ["visualizations", "charts"]}
 *
 * Bar charts are great for showing data that fits neatly in to "buckets". The
 * x axis data must be strings, and the y axis data must be numeric.
 */
const BarChart = createClass({
	displayName: 'BarChart',

	propTypes: {
		/**
		 * Classes are appended to root element along with existing classes using
		 * the `classnames` library.
		 */
		className: any,
		/**
		 * Height of the chart.
		 */
		height: number,
		/**
		 * Width of the chart.
		 */
		width: number,
		/**
		 * An object defining the margins of the chart. These margins typically
		 * contain the axis and labels.
		 */
		margin: shape({
			top: number,
			right: number,
			bottom: number,
			left: number,
		}),
		/**
		 * Data for the chart. E.g.
		 *
		 *     [
		 *       { x: 'Monday'    , y: 1 } ,
		 *       { x: 'Tuesday'   , y: 2 } ,
		 *       { x: 'Wednesday' , y: 3 } ,
		 *       { x: 'Thursday'  , y: 2 } ,
		 *       { x: 'Friday'    , y: 5 } ,
		 *     ]
		 */
		data: arrayOf(object).isRequired,
		/**
		 * An object with human readable names for fields that  will be used for
		 * tooltips and legends which are *not yet implemented*. E.g:
		 *
		 *     {
		 *       x: 'Revenue',
		 *       y: 'Impressions',
		 *     }
		 *
		 * legend: object,
		 */


		/**
		 * The field we should look up your x data by. Your actual x data must be
		 * strings.
		 */
		xAxisField: string,
		/**
		 * There are some cases where you need to only show a "sampling" of ticks
		 * on the x axis. This number will control that.
		 */
		xAxisTickCount: number,
		/**
		 * An optional function used to format your x axis data. If you don't
		 * provide anything, we'll use an identity function.
		 */
		xAxisFormatter: func,
		/**
		 * Set a title for the x axis.
		 */
		xAxisTitle: string,
		/**
		 * Set a color for the x axis title. This takes any number 0 or greater and
		 * it converts it to a color in our color palette.
		 */
		xAxisTitleColor: number,


		/**
		 * An array of your y axis fields. Typically this will just be a single
		 * item unless you need to display grouped or stacked bars.
		 */
		yAxisFields: array.isRequired,
		/**
		 * The minimum number the y axis should display. Typically this
		 * should be be `0`.
		 */
		yAxisMin: number,
		/**
		 * The maximum number the y axis should display. This should almost always
		 * be the largest number from your dataset.
		 */
		yAxisMax: number,
		/**
		 * An optional function used to format your y axis data. If you don't
		 * provide anything, we use the default D3 number formatter.
		 */
		yAxisFormatter: func,
		/**
		 * Stack the y axis data instead of showing it as groups. This is only
		 * useful if you have multiple `yAxisFields`. Stacking will cause the chart
		 * to be aggregated by sum.
		 */
		yAxisIsStacked: bool,
		/**
		 * There are some cases where you need to only show a "sampling" of ticks
		 * on the y axis. This number will control that.
		 */
		yAxisTickCount: number,
		/**
		 * Set a title for the y axis.
		 */
		yAxisTitle: string,
		/**
		 * Set a color for the y axis title. This takes any number 0 or greater and
		 * it converts it to a color in our color palette.
		 */
		yAxisTitleColor: number,
	},

	getDefaultProps() {
		return {
			height: 400,
			width: 1000,
			margin: {
				top: 10,
				right: 20,
				bottom: 50,
				left: 80,
			},

			xAxisField: 'x',
			xAxisTickCount: null,
			xAxisTitle: null,
			xAxisTitleColor: -1,

			yAxisFields: ['y'],
			yAxisTickCount: null,
			yAxisIsStacked: false,
			yAxisMin: 0,
			yAxisTitle: null,
			yAxisTitleColor: -1,
		};
	},

	render() {
		const {
			className,
			height,
			width,
			margin,
			data,

			xAxisField,
			xAxisFormatter,
			xAxisTitle,
			xAxisTitleColor,
			xAxisTickCount,

			yAxisFields,
			yAxisFormatter,
			yAxisTitle,
			yAxisTitleColor,
			yAxisIsStacked,
			yAxisTickCount,
			yAxisMin,
			yAxisMax = yAxisIsStacked
				? maxByFieldsStacked(data, yAxisFields)
				: maxByFields(data, yAxisFields),

			...passThroughs,
		} = this.props;

		if (_.isEmpty(data)) {
			return null;
		}

		const innerWidth = width - margin.left - margin.right;
		const innerHeight = height - margin.top - margin.bottom;

		const paddingInner = yAxisFields.length > 1 ? 0.3 : 0.05;

		const xScale = d3Scale.scaleBand()
			.domain(_.map(data, xAxisField))
			.range([0, innerWidth])
			.paddingInner(paddingInner)
			.paddingOuter(0.5);

		const yScale = d3Scale.scaleLinear()
			.domain([yAxisMin, yAxisMax])
			.range([innerHeight, 0]);

		return (
			<svg
				{...passThroughs}
				className={cx(className, '&')}
				width={width}
				height={height}
			>
				{/* x axis */}
				<g transform={`translate(${margin.left}, ${innerHeight + margin.top})`}>
					<Axis
						orient='bottom'
						scale={xScale}
						outerTickSize={0}
						tickCount={xAxisTickCount}
						tickFormat={xAxisFormatter}
					/>
				</g>

				{/* x axis title */}
				<g transform={`translate(${margin.left}, ${margin.top + innerHeight})`}>
					{xAxisTitle ? (
						<AxisLabel
							orient='bottom'
							width={innerWidth}
							height={margin.bottom}
							label={xAxisTitle}
							color={xAxisTitleColor}
						/>
					) : null}
				</g>

				{/* y axis */}
				<g transform={`translate(${margin.left}, ${margin.top})`}>
					<Axis
						orient='left'
						scale={yScale}
						tickFormat={yAxisFormatter}
						tickCount={yAxisTickCount}
					/>
				</g>

				{/* y axis title */}
				<g transform={`translate(0, ${margin.top})`}>
					{yAxisTitle ? (
						<AxisLabel
							orient='left'
							width={margin.left}
							height={innerHeight}
							label={yAxisTitle}
							color={yAxisTitleColor}
						/>
					) : null}
				</g>

				{/* bars */}
				<Bars
					top={margin.top}
					left={margin.left}
					xField={xAxisField}
					xScale={xScale}
					yFields={yAxisFields}
					yScale={yScale}
					data={data}
					isStacked={yAxisIsStacked}
				/>
			</svg>
		);
	}
});

export default BarChart;
