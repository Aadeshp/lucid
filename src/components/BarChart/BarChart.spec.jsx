// Note: these tests are basically pin tests, given that we're rendering svgs,
// these tests serve to ensure that the rendered output is exactly at the author
// intended. As a consequence, you may need to re-pin these tests if you change
// things.

import React from 'react';
import { mount } from 'enzyme';
import { common } from '../../util/generic-tests';
import describeWithDOM from '../../util/describe-with-dom';
import assert from 'assert';

import BarChart from './BarChart';

describeWithDOM('BarChart', () => {
	let wrapper;

	afterEach(() => {
		if (wrapper) {
			wrapper.unmount();
		}
	});

	common(BarChart, {
		exemptFunctionProps: [
			'xAxisFormatter',
			'yAxisFormatter',
		],
		getDefaultProps: () => ({
			data: [
				{ x: 'Monday'    , y: 1  , y2: 2}   ,
				{ x: 'Tuesday'   , y: 4  , y2: 4 }  ,
				{ x: 'Wednesday' , y: 8  , y2: 1 }  ,
				{ x: 'Thursday'  , y: 20 , y2: 15 } ,
				{ x: 'Friday'    , y: 10 , y2: 2 }  ,
			],
		}),
	});

	describe('render', () => {
		it('should render a basic chart', () => {
			wrapper = mount(
				<BarChart
					data={[
						{ x: 'Monday'    , y: 1  , y2: 2}   ,
						{ x: 'Tuesday'   , y: 4  , y2: 4 }  ,
						{ x: 'Wednesday' , y: 8  , y2: 1 }  ,
						{ x: 'Thursday'  , y: 20 , y2: 15 } ,
						{ x: 'Friday'    , y: 10 , y2: 2 }  ,
					]}
				/>
			);

			assert.equal(wrapper.find('.lucid-Bar').length, 5, 'did not find the correct number of bars');
			assert.equal(wrapper.find('.lucid-Axis').length, 2, 'did not find the correct number of axes');
		});

		it('should render a chart with multiple series', () => {
			wrapper = mount(
				<BarChart
					data={[
						{ x: 'Monday'    , y: 1  , y2: 2}   ,
						{ x: 'Tuesday'   , y: 4  , y2: 4 }  ,
						{ x: 'Wednesday' , y: 8  , y2: 1 }  ,
						{ x: 'Thursday'  , y: 20 , y2: 15 } ,
						{ x: 'Friday'    , y: 10 , y2: 2 }  ,
					]}
					yAxisFields={['y', 'y2']}
				/>
			);

			assert.equal(wrapper.find('.lucid-Bar').length, 10, 'did not find the correct number of bars');
		});

		it('should have the correct html', () => {
			wrapper = mount(
				<BarChart
					height={500}
					width={1000}
					margins={{
						top: 50,
						right: 50,
						bottom: 50,
						left: 50,
					}}
					data={[
						{date: '2015-01-01', rev: 100, imps: 1000},
						{date: '2015-01-02', rev: 89, imps: 2200},
						{date: '2015-01-03', rev: 95, imps: 3305},
					]}

					xAxisField='date'
					xAxisTitle='Date'
					xAxisTickCount={2}
					xAxisFormatter={() => 'x axis tick'}

					yAxisFields={['rev', 'imps']}
					yAxisTitle='Metrics'
					yAxisTickCount={4}
					yAxisFormatter={() => 'y axis tick'}
				/>
			);

			assert.equal(wrapper.html(), '<svg class="lucid-BarChart" width="1000" height="500"><g transform="translate(80, 450)"><g class="lucid-Axis"><path class="lucid-Axis-domain" d="M0,0V0H900V0"></path><g transform="translate(206.75675675675674, 0)"><line class="lucid-Axis-tick" x2="0" y2="6"></line><text class="lucid-Axis-tick-text" x="0" y="9" dy=".71em" style="text-anchor:middle;">2015-01-01</text></g><g transform="translate(693.2432432432432, 0)"><line class="lucid-Axis-tick" x2="0" y2="6"></line><text class="lucid-Axis-tick-text" x="0" y="9" dy=".71em" style="text-anchor:middle;">2015-01-03</text></g></g></g><g transform="translate(80, 450)"><text style="fill:#000;" class="lucid-AxisLabel" x="450" y="50" dy="-.32em" transform="">Date</text></g><g transform="translate(80, 10)"><g class="lucid-Axis"><path class="lucid-Axis-domain" d="M-6,440H0V0H-6"></path><g transform="translate(0, 440)"><line class="lucid-Axis-tick" x2="-6" y2="0"></line><text class="lucid-Axis-tick-text" x="-9" y="0" dy=".32em" style="text-anchor:end;">y axis tick</text></g><g transform="translate(0, 306.86838124054464)"><line class="lucid-Axis-tick" x2="-6" y2="0"></line><text class="lucid-Axis-tick-text" x="-9" y="0" dy=".32em" style="text-anchor:end;">y axis tick</text></g><g transform="translate(0, 173.73676248108927)"><line class="lucid-Axis-tick" x2="-6" y2="0"></line><text class="lucid-Axis-tick-text" x="-9" y="0" dy=".32em" style="text-anchor:end;">y axis tick</text></g><g transform="translate(0, 40.60514372163391)"><line class="lucid-Axis-tick" x2="-6" y2="0"></line><text class="lucid-Axis-tick-text" x="-9" y="0" dy=".32em" style="text-anchor:end;">y axis tick</text></g></g></g><g transform="translate(0, 10)"><text style="fill:#000;" class="lucid-AxisLabel" x="-220" y="0" dy="1em" transform="rotate(-90)">Metrics</text></g><g transform="translate(80, 10)"><g class="lucid-Bars"><g><rect class="lucid-Bar lucid-Bar-color-chart-0" x="121.62162162162161" y="426.6868381240545" height="13.313161875945525" width="85"></rect><rect class="lucid-Bar lucid-Bar-color-chart-1" x="206.6216216216216" y="306.86838124054464" height="133.13161875945536" width="85"></rect><g class="lucid-ContextMenu lucid-ToolTip"><rect class="lucid-Bars-tooltip-hover-zone" height="133.13161875945536" width="170.27027027027026" x="121.62162162162161" y="306.86838124054464"></rect></g></g><g><rect class="lucid-Bar lucid-Bar-color-chart-0" x="364.86486486486484" y="428.15128593040845" height="11.848714069591551" width="85"></rect><rect class="lucid-Bar lucid-Bar-color-chart-1" x="449.86486486486484" y="147.11043872919822" height="292.8895612708018" width="85"></rect><g class="lucid-ContextMenu lucid-ToolTip"><rect class="lucid-Bars-tooltip-hover-zone" height="292.8895612708018" width="170.27027027027026" x="364.86486486486484" y="147.11043872919822"></rect></g></g><g><rect class="lucid-Bar lucid-Bar-color-chart-0" x="608.1081081081081" y="427.35249621785175" height="12.647503782148249" width="85"></rect><rect class="lucid-Bar lucid-Bar-color-chart-1" x="693.1081081081081" y="0" height="440" width="85"></rect><g class="lucid-ContextMenu lucid-ToolTip"><rect class="lucid-Bars-tooltip-hover-zone" height="440" width="170.27027027027026" x="608.1081081081081" y="0"></rect></g></g></g></g></svg>');
		});
	});
});

