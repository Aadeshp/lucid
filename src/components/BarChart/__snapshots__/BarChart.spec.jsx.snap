// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart [common] example testing should match snapshot(s) for 01.basic 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Revenue"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "x": "2015-01-01",
            "y": 1,
          },
          Object {
            "x": "2015-01-02",
            "y": 2,
          },
          Object {
            "x": "2015-01-03",
            "y": 3,
          },
          Object {
            "x": "2015-01-04",
            "y": 5,
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "y",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={5}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 03.grouped 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Fruit Count"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "apples": 10,
            "bananas": 15,
            "oranges": 5,
            "peaches": 35,
            "pears": 20,
            "x": "Monday",
          },
          Object {
            "apples": 20,
            "bananas": 25,
            "oranges": 27,
            "peaches": 20,
            "pears": 5,
            "x": "Tuesday",
          },
          Object {
            "apples": 5,
            "bananas": 20,
            "oranges": 35,
            "peaches": 5,
            "pears": 15,
            "x": "Wednesday",
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0-lightest",
          "color-chart-0-light",
          "color-chart-0",
          "color-chart-0-dark",
          "color-chart-0-darkest",
          "color-chart-1-lightest",
          "color-chart-1-light",
          "color-chart-1",
          "color-chart-1-dark",
          "color-chart-1-darkest",
          "color-chart-2-lightest",
          "color-chart-2-light",
          "color-chart-2",
          "color-chart-2-dark",
          "color-chart-2-darkest",
          "color-chart-3-lightest",
          "color-chart-3-light",
          "color-chart-3",
          "color-chart-3-dark",
          "color-chart-3-darkest",
          "color-chart-4-lightest",
          "color-chart-4-light",
          "color-chart-4",
          "color-chart-4-dark",
          "color-chart-4-darkest",
          "color-chart-5-lightest",
          "color-chart-5-light",
          "color-chart-5",
          "color-chart-5-dark",
          "color-chart-5-darkest",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "apples",
          "pears",
          "peaches",
          "bananas",
          "oranges",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={35}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 04.grouped-with-legend 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Fruit Count"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "apples": 10,
            "bananas": 15,
            "oranges": 5,
            "peaches": 35,
            "pears": 20,
            "x": "Monday",
          },
          Object {
            "apples": 20,
            "bananas": 25,
            "oranges": 27,
            "peaches": 20,
            "pears": 5,
            "x": "Tuesday",
          },
          Object {
            "apples": 5,
            "bananas": 20,
            "oranges": 35,
            "peaches": 5,
            "pears": 15,
            "x": "Wednesday",
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "apples",
          "pears",
          "peaches",
          "bananas",
          "oranges",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={35}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 05.limited-ticks 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={5}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={4}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "x": "2015-01-01",
            "y": 1,
          },
          Object {
            "x": "2015-01-02",
            "y": 2,
          },
          Object {
            "x": "2015-01-03",
            "y": 3,
          },
          Object {
            "x": "2015-01-04",
            "y": 5,
          },
          Object {
            "x": "2015-01-05",
            "y": 2,
          },
          Object {
            "x": "2015-01-06",
            "y": 3,
          },
          Object {
            "x": "2015-01-07",
            "y": 2,
          },
          Object {
            "x": "2015-01-08",
            "y": 2,
          },
          Object {
            "x": "2015-01-09",
            "y": 5,
          },
          Object {
            "x": "2015-01-10",
            "y": 3,
          },
          Object {
            "x": "2015-01-11",
            "y": 4,
          },
          Object {
            "x": "2015-01-12",
            "y": 4,
          },
          Object {
            "x": "2015-01-13",
            "y": 5,
          },
          Object {
            "x": "2015-01-14",
            "y": 3,
          },
          Object {
            "x": "2015-01-15",
            "y": 4,
          },
          Object {
            "x": "2015-01-16",
            "y": 3,
          },
          Object {
            "x": "2015-01-17",
            "y": 6,
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "y",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={6}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 06.all-the-things 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={2}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 350)"
  >
    <AxisLabel
      color="#000"
      height={50}
      label="Weekdays"
      orient="bottom"
      width={900}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={4}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Fruit Count"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorMap={
        Object {
          "apples": "color-chart-good",
          "oranges": "color-chart-1",
        }
      }
      colorOffset={0}
      data={
        Array [
          Object {
            "apples": 2000,
            "day": "monday",
            "oranges": 3000,
          },
          Object {
            "apples": 2000,
            "day": "tuesday",
            "oranges": 5000,
          },
          Object {
            "apples": 3000,
            "day": "wednesday",
            "oranges": 2000,
          },
          Object {
            "apples": 5000,
            "day": "thursday",
            "oranges": 6000,
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="day"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "apples",
          "oranges",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={6000}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 07.stacked 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Fruit Count"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "apples": 10,
            "peaches": 35,
            "pears": 20,
            "x": "Monday",
          },
          Object {
            "apples": 20,
            "peaches": 20,
            "pears": 5,
            "x": "Tuesday",
          },
          Object {
            "apples": 5,
            "peaches": 5,
            "pears": 15,
            "x": "Wednesday",
          },
        ]
      }
      hasToolTips={true}
      isStacked={true}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "apples",
          "pears",
          "peaches",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={65}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 08.unformatted-tooltips 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Revenue"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "x": "2015-01-01",
            "y": 1200,
          },
          Object {
            "x": "2015-01-02",
            "y": 900,
          },
          Object {
            "x": "2015-01-03",
            "y": 1800,
          },
          Object {
            "x": "2015-01-04",
            "y": 3000,
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "y",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={3000}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 09.formatted-tooltips 1`] = `
<svg
  className="lucid-BarChart"
  height={400}
  width={1000}
>
  <g
    transform="translate(80, 350)"
  >
    <Axis
      innerTickSize={6}
      orient="bottom"
      outerTickSize={0}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Axis
      innerTickSize={6}
      orient="left"
      outerTickSize={6}
      scale={[Function]}
      tickCount={null}
      tickFormat={[Function]}
      tickPadding={3}
    />
  </g>
  <g
    transform="translate(0, 10)"
  >
    <AxisLabel
      color="#000"
      height={340}
      label="Revenue"
      orient="left"
      width={80}
    />
  </g>
  <g
    transform="translate(80, 10)"
  >
    <Bars
      colorOffset={0}
      data={
        Array [
          Object {
            "x": "2015-01-01",
            "y": 1200,
          },
          Object {
            "x": "2015-01-02",
            "y": 900,
          },
          Object {
            "x": "2015-01-03",
            "y": 1800,
          },
          Object {
            "x": "2015-01-04",
            "y": 3000,
          },
        ]
      }
      hasToolTips={true}
      isStacked={false}
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      xField="x"
      xFormatter={[Function]}
      xScale={[Function]}
      yFields={
        Array [
          "y",
        ]
      }
      yFormatter={[Function]}
      yScale={[Function]}
      yStackedMax={3000}
      yTooltipFormatter={[Function]}
    />
  </g>
</svg>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 10.empty 1`] = `
<EmptyStateWrapper
  Title="You have no data."
  isEmpty={true}
>
  <svg
    className="lucid-BarChart"
    height={400}
    width={1000}
  >
    <g
      transform="translate(80, 350)"
    >
      <Axis
        innerTickSize={6}
        orient="bottom"
        outerTickSize={6}
        scale={[Function]}
        tickCount={null}
        tickPadding={3}
      />
    </g>
    <g
      transform="translate(80, 10)"
    >
      <Axis
        innerTickSize={6}
        orient="left"
        outerTickSize={6}
        scale={[Function]}
        tickCount={null}
        tickFormat={[Function]}
        tickPadding={3}
      />
    </g>
  </svg>
</EmptyStateWrapper>
`;

exports[`BarChart [common] example testing should match snapshot(s) for 11.empty-with-custom-title-and-body 1`] = `
<EmptyStateWrapper
  isEmpty={true}
>
  <EmptyStateWrapper.Title>
    Something went wrong.
  </EmptyStateWrapper.Title>
  <EmptyStateWrapper.Body
    style={
      Object {
        "fontSize": "12px",
      }
    }
  >
    Echo park poutine esse tempor squid do. Lo-fi ramps XOXO chicharrones laboris, portland fugiat locavore. Fap four dollar toast keytar, cronut kogi fingerstache distillery microdosing everyday carry austin DIY dreamcatcher. Distillery flexitarian meditation laboris roof party. Cred raclette gastropub tilde PBR&B. Shoreditch poke adipisicing, reprehenderit lumbersexual succulents mustache officia franzen vinyl nostrud af. Hashtag bitters organic, before they sold out butcher cronut sapiente.
  </EmptyStateWrapper.Body>
  <svg
    className="lucid-BarChart"
    height={400}
    width={1000}
  >
    <g
      transform="translate(80, 350)"
    >
      <Axis
        innerTickSize={6}
        orient="bottom"
        outerTickSize={6}
        scale={[Function]}
        tickCount={null}
        tickPadding={3}
      />
    </g>
    <g
      transform="translate(80, 10)"
    >
      <Axis
        innerTickSize={6}
        orient="left"
        outerTickSize={6}
        scale={[Function]}
        tickCount={null}
        tickFormat={[Function]}
        tickPadding={3}
      />
    </g>
  </svg>
</EmptyStateWrapper>
`;
